Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts

```diff
@@ -1,23 +1,86 @@
-  var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
-  var allFiles = tl.find(buildFolder);
-  var codeCoverageFiles = tl.match(allFiles, additionalCodeCoverageFiles, { matchBase: true });
+        // Resolve the summary file path.
+        // It may contain wildcards allowing the path to change between builds, such as for:
+        // $(System.DefaultWorkingDirectory)\artifacts***$(Configuration)\testresults\coverage\cobertura.xml
+        var resolvedSummaryFile: string = resolvePathToSingleItem(workingDirectory, summaryFileLocation);
+
+        // Resolve the report directory.
+        // It may contain wildcards allowing the path to change between builds, such as for:
+        // $(System.DefaultWorkingDirectory)\artifacts***$(Configuration)\testresults\coverage
+        var resolvedReportDirectory: string = resolvePathToSingleItem(workingDirectory, reportDirectory);
+
+        // Get any 'Additional Files' to publish as build artifacts
+        if (additionalFiles) {
+            // Does the 'Additional Files' value contain wildcards?
+            if (containsWildcard(additionalFiles)) {
+                // Resolve matches of the 'Additional Files' pattern
+                var additionalFileMatches: string[] = tl.findMatch(workingDirectory, additionalFiles, null, { matchBase: true });
+                tl.debug(tl.loc('FoundNMatchesForPattern', additionalFileMatches.length, additionalFiles));
+            }
+            else {
+                // Use the specific additional file (no wildcards)
+                var additionalFileMatches: string[] = [additionalFiles];
+            }
+        }
+
+        // Publish code coverage data
+        var ccPublisher = new tl.CodeCoveragePublisher();
+        ccPublisher.publish(codeCoverageTool, resolvedSummaryFile, resolvedReportDirectory, additionalFileMatches);
+    }
+    catch (err) {
+        tl.setResult(tl.TaskResult.Failed, err);
+    }
+}
+
+// Resolves the specified path to a single item based on whether it contains wildcards
+function resolvePathToSingleItem(workingDirectory:string, pathInput: string) : string {
+    // Default to using the specific pathInput value
+    var resolvedPath: string = pathInput;
+
+    // Does the pathInput value contain wildcards?
+    if (pathInput && containsWildcard(pathInput)) {
+        // Resolve matches of the pathInput pattern
+        var pathMatches: string[] = tl.findMatch(workingDirectory, pathInput);
+        tl.debug(tl.loc('FoundNMatchesForPattern', pathMatches.length, pathInput));
+
+        // Were any matches found?
+        if (pathMatches.length == 0) {
+            resolvedPath = undefined;
+        }
+        else {
+            // Select the path to be used from the matches
+            resolvedPath = pathMatches[0];
+
+            // If more than one path matches, use the first and issue a warning
+            if (pathMatches.length > 1) {
+                tl.warning(tl.loc('MultipleSummaryFilesFound', resolvedPath));
```

Comment:
This is called for both a file and a directory.  This is the error message: "Multiple matching code coverage summary files were found. Using the first match: %s".  Should we change this to be more generic for files and folders?